README_cmake.md

```diff
@@ -750,6 +750,8 @@ Variables that control package loading:
 | Variable             | Description                                                                                                                                                                   |
 |----------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
 | `Halide_SHARED_LIBS` | override `BUILD_SHARED_LIBS` when loading the Halide package via `find_package`. Has no effect when using Halide via `add_subdirectory` as a Git or `FetchContent` submodule. |
+| `Halide_RUNTIME_NO_THREADS` | skip linking of Threads libraray to runtime. Should be set if your toolchain does not support it (e.g. baremetal). |
+| `Halide_RUNTIME_NO_DL_LIBS` | skip linking of DL libraray to runtime. Should be set if your toolchain does not support it (e.g. baremetal). |
```

Comment:
Can you sketch an example where unconditionally linking to `${CMAKE_DL_LIBS}` is a problem? My understanding is that it is very often empty, and indeed should be empty when the target platform does not support it.

Reply:
In `HelloBaremetal` sample app, `dl` is set to `CMAKE_DL_LIBS`. This doesn't change even if `set(CMAKE_DL_LIBS "")` in toolchain file. It seems to be overwritten in [CMakeGenericSystem.cmake](https://github.com/Kitware/CMake/blob/c5a945dce8e1a4fb1b1db91c2dae03795c74074b/Modules/CMakeGenericSystem.cmake#L22)

Reply:
@stevesuzuki-arm -- can you try one thing for me?

Try setting `CMAKE_SYSTEM_NAME` in the HelloBaremetal app to `Generic-ELF` instead of `none`. If that works, let's remove the `Halide_RUNTIME_NO_DL_LIBS` control

Reply:
Okay, I've spoken with the CMake devs. Ben Boeckel thinks CMake is doing the wrong thing here, but fixing it would require a policy. So this is fine for now.

Reply:
That's good to know. JFYI, I tried `Generic` and `Generic-ELF` but neither of them solved DL issue.

Reply:
Thanks... sorry about the comment churn.